api.planx.uk/modules/flows/updateTemplatedFlow/service.ts

```diff
@@ -0,0 +1,97 @@
+import type { FlowGraph } from "@opensystemslab/planx-core/types";
+import { gql } from "graphql-request";
+import merge from "lodash/merge.js";
+import { $api } from "../../../client/index.js";
+import { getFlowData } from "../../../helpers.js";
+import type { Flow } from "../../../types.js";
+
+interface GetTemplatedFlowEdits {
+  edits: {
+    data: Flow["data"]; // Not `FlowGraph` because not required to have `_root` !
+  }[];
+}
+
+interface UpdateTemplatedFlowData {
+  flow: {
+    data: FlowGraph;
+  };
+}
+
+interface InsertComment {
+  comment: {
+    id: number;
+  };
+}
+
+export const updateTemplatedFlow = async (
+  sourceFlowId: string,
+  templatedFlowId: string,
+) => {
+  const { data: sourceData, publishedFlows } = await getFlowData(sourceFlowId);
+  const { publisher_id: sourcePublisher, summary } = publishedFlows[0];
+
+  // Get templated flow edits (aka "customisations")
+  //   Since initiated via a Hasura event, we don't have user context here and rely on $api client
+  const templatedFlowEditsResponse =
+    await $api.client.request<GetTemplatedFlowEdits>(
+      gql`
+        query GetTemplatedFlowEdits($flow_id: uuid!) {
+          edits: templated_flow_edits(where: { flow_id: { _eq: $flow_id } }) {
+            data
+          }
+        }
+      `,
+      {
+        flow_id: templatedFlowId,
+      },
+    );
+  const edits = templatedFlowEditsResponse.edits[0].data;
+
+  // Apply templated flow edits on top of source data using Lodash's deep merge
+  const data = merge(sourceData, edits);
```

Comment:
This is satifyingly simple and feels reassuring that using a jsonb column type for `templated_flow_edits.data` is the right call ! 

Lodash's `merge` is doing exactly what I expect here so far - and if we find out we need more flexibilty then we can look into `mergeWith` instead which offers a `customizer` param ([docs](https://lodash.com/docs/4.17.15#mergeWith))